Connecting to localhost rather than configured registry address

Hi,

I'm running marathon-consul as a docker instance deployed by puppet, here is the process:
```
CONTAINER ID                                                       IMAGE                        COMMAND                                                                                           CREATED             STATUS              PORTS                                                                                                                                                        NAMES
a9adf6b884bd3475211305f049082cd907447fdee890d36467b5609369541cb7   ciscocloud/marathon-consul   "/launch.sh --marathon-location=172.17.42.1:8080 --registry=172.17.42.1:8500 --log-level=debug"   23 minutes ago      Up 23 minutes       0.0.0.0:4000->4000/tcp                                                                                                                                       marathon-consul
```

As you can see, I've specified the registry and marathon locations to be the docker bridge IP.

Then I start marathon with the callback flag and create a callback:
```
curl -X POST 'http://localhost:8080/v2/eventSubscriptions?callbackUrl=http://localhost:4000/events' -v
```

I've also tried creating a callback with docker bridge IP:
```
curl -X POST 'http://172.17.42.1:8080/v2/eventSubscriptions?callbackUrl=http://172.17.42.1:4000/events' -v
```

In both instances when running `marathon-consul` in debug mode, I see the following:
```
time="2015-10-18T17:28:22Z" level=info msg="handling event" eventType="status_update_event" 
time="2015-10-18T17:28:22Z" level=debug msg="{\"slaveId\":\"20151017-195725-3111233728-5050-17500-S0\",\"taskId\":\"basic-0.838db8f8-75bd-11e5-b38e-5e84bfba7cb5\",\"taskStatus\":\"TASK_KILLED\",\"message\":\"Command terminated with signal Terminated\",\"appId\":\"/basic-0\",\"host\":\"mesos0\",\"ports\":[20777],\"version\":\"2015-10-18T17:27:27.950Z\",\"eventType\":\"status_update_event\",\"timestamp\":\"2015-10-18T17:28:22.309Z\"}" 
time="2015-10-18T17:29:18Z" level=info msg="handling event" eventType="api_post_event" 
time="2015-10-18T17:29:18Z" level=info msg="[ERROR] response generated error: Get http://127.0.0.1:8500/v1/kv/marathon/basic-0: dial tcp 127.0.0.1:8500: connection refused" 
time="2015-10-18T17:29:18Z" level=debug msg="{\"clientIp\":\"10.2.9.151\",\"uri\":\"/v2/apps//basic-0\",\"appDefinition\":{\"id\":\"/basic-0\",\"cmd\":null,\"args\":null,\"user\":null,\"env\":{},\"instances\":1,\"cpus\":1.0,\"mem\":128.0,\"disk\":0.0,\"executor\":\"\",\"constraints\":[],\"uris\":[],\"storeUrls\":[],\"ports\":[0],\"requirePorts\":false,\"backoffSeconds\":1,\"backoffFactor\":1.15,\"maxLaunchDelaySeconds\":3600,\"container\":null,\"healthChecks\":[],\"dependencies\":[],\"upgradeStrategy\":{\"minimumHealthCapacity\":1.0,\"maximumOverCapacity\":1.0},\"labels\":{},\"acceptedResourceRoles\":null,\"version\":\"2015-10-18T17:29:18.558Z\"},\"eventType\":\"api_post_event\",\"timestamp\":\"2015-10-18T17:29:18.560Z\"}" 
time="2015-10-18T17:29:18Z" level=info msg="not handling event" eventType="group_change_success" 
```

I can't seem to work out why the query is going to '127.0.0.1:8500' rather than the configured registry address. Any help would be appreciated!

Cheers,

```
root@mesos0:~# dpkg -l | grep -E 'marathon|mesos'
ii  marathon                            0.9.0-1.0.381.debian77            amd64        Cluster-wide init and control system for services running on Apache Mesos
ii  mesos                               0.22.1-1.0.debian78               amd64        Cluster resource manager with efficient resource isolation
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,6 @@
 language: go
 
 go:
-  - 1.4
   - 1.5
   - tip
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,8 @@
 language: go
 
 go:
-  - 1.3
+  - 1.4
+  - 1.5
   - tip
 
 install: make deps

--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ build: deps
 	go build -o bin/$(NAME)
 
 test: deps
-	go test $(TEST) $(TESTARGS) -parallel=4
+	go test $(TEST) $(TESTARGS)
 	go vet $(TEST)
 
 xcompile: deps test

--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ build: deps
 	go build -o bin/$(NAME)
 
 test: deps
-	go test $(TEST) $(TESTARGS) -timeout=30s -parallel=4
+	go test $(TEST) $(TESTARGS) -parallel=4
 	go vet $(TEST)
 
 xcompile: deps test

--- a/config/config.go
+++ b/config/config.go
@@ -13,6 +13,7 @@ import (
 
 var (
 	ErrBadCredentials = errors.New("credentials must be of the form `user:pass`")
+	ErrNoScheme       = errors.New("please specify a scheme for the registry")
 )
 
 type Config struct {
@@ -98,6 +99,9 @@ func (r Registry) Config() (*api.Config, error) {
 	url, err := url.Parse(r.Location)
 	if err != nil {
 		return nil, err
+	}
+	if url.Scheme == "" {
+		return nil, ErrNoScheme
 	}
 
 	auth, err := r.GetAuth()

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -32,3 +32,12 @@ func TestGetAuth(t *testing.T) {
 	assert.Nil(t, auth)
 	assert.Equal(t, ErrBadCredentials, err)
 }
+
+func TestRegistryConfigURL(t *testing.T) {
+	t.Parallel()
+
+	reg := Registry{Location: "x"}
+	c, err := reg.Config()
+	assert.Nil(t, c)
+	assert.Equal(t, err, ErrNoScheme)
+}

